modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -221,6 +224,65 @@ public void processOpts() {
         }
     }
 
+    @Override
+    public String toModelImport( String name){
+        if(name.contains("|")){
+            List<String> names = Arrays.asList(name.split("\\|"));
+           return names.stream()
+                   .map(withSpace->withSpace.replaceAll(" ",""))
+                   .map(noSpace->super.toModelImport(noSpace))
+                   .collect(Collectors.joining("|"));
+        }
+        return super.toModelImport(name);
+    }
+
+    protected String toModelImportForUnionTypes(String name, Function<String,String> toModelImportSingle){
+        if(name.contains("|")){
+            List<String> names = Arrays.asList(name.split("\\|"));
+            return names.stream()
+                    .map(withSpace->withSpace.replaceAll(" ",""))
+                    .map(noSpace->toModelImportSingle.apply(noSpace))
+                    .collect(Collectors.joining("|"));
+        }
+        return toModelImportSingle.apply(name);
+    }
+
+    @Override
+    public Map<String, Object> postProcessOperationsWithModels(final Map<String, Object> objs, final List<Object> allModels) {
+        Map<String, Object> objsUnderProcess=  super.postProcessOperationsWithModels(objs, allModels);
+
+        if(Boolean.valueOf(objsUnderProcess.get("hasImport").toString())){
+            objsUnderProcess.put("imports", splitUniontypeImports((List<Map<String,String>>) objsUnderProcess.get("imports")));
+        }
+        return objsUnderProcess;
+    }
+
+    protected List<Object> splitUniontypeImports(List<Map<String,String>> imports){
```

Comment:
@amakhrov @wing328  do you think such a complicated logic is necessary, or could this be solved elsewhere?

Reply:
It looks quite similar to the existing logic in TypeScriptAngularClientCodegen (see `postProcessAllModels` and `parseImports`)

Maybe we should lift the logic from AngularClient to AbstractTypescriptClient instead of pretty much repeating it?

The root problem, as I see it, is that `toModelImport` takes a string as an argument rather than a model and returns a string instead of a list. It makes it very limited in terms of handling composite models. But changing that would require a change in Generator and CodegenConfig interfaces. And generally means more work, of course.